app/controllers/recipes_controller.rb

```diff
@@ -72,4 +73,16 @@ def recipe_params
       params.require(:recipe).permit(:name, :description, :preparation_time,
                                       recipe_products_attributes: [:id, :recipe_id, :product_id, :quantity, :_destroy])
     end
+
+    def fetch_product_calories
+      @recipe.products.each do |product|
+        if !!(ProductCalories.exists?(product.id))
```

Comment:
it doesn't seem like a good idea to create new instance of ApiClient in each iteration and do request in such loop. I think this logic would suit product model better, so each time product is created/updated then call to api would be made